api/notd/token_staking_processor.py

```diff
@@ -64,35 +64,32 @@ async def retrieve_updated_token_staking(self, registryAddress: str, tokenId: st
-    async def retrieve_token_stakings(self, registryAddress: str) -> List[RetrievedTokenStaking]:
-        retrievedStakedTokens: List[RetrievedTokenStaking] = []
-        if registryAddress != COLLECTION_CREEPZ_ADDRESS:
-            return retrievedStakedTokens
-        for stakingAddress in STAKING_ADDRESSES:
-            stakedQuery = (
-                sqlalchemy.select(TokenTransfersTable.c.tokenId, TokenTransfersTable.c.fromAddress, TokenTransfersTable.c.transactionHash, BlocksTable.c.blockDate)
-                .join(BlocksTable, BlocksTable.c.blockNumber == TokenTransfersTable.c.blockNumber)
-                .where(TokenTransfersTable.c.registryAddress == registryAddress)
-                .where(TokenTransfersTable.c.toAddress == stakingAddress)
-                .order_by(TokenTransfersTable.c.blockNumber.asc())
-            )
-            stakedTokensResult = await self.retriever.database.execute(query=stakedQuery)
-            stakedTokens = list(stakedTokensResult)
-            unStakedQuery = (
-                sqlalchemy.select(TokenTransfersTable.c.tokenId, TokenTransfersTable.c.toAddress, TokenTransfersTable.c.transactionHash, BlocksTable.c.blockDate)
-                .join(BlocksTable, BlocksTable.c.blockNumber == TokenTransfersTable.c.blockNumber)
-                .where(TokenTransfersTable.c.registryAddress == registryAddress)
-                .where(TokenTransfersTable.c.fromAddress == stakingAddress)
-                .order_by(TokenTransfersTable.c.blockNumber.asc())
-            )
-            unStakedTokensResult = await self.retriever.database.execute(query=unStakedQuery)
-            unStakedTokens = list(unStakedTokensResult)
-            currentlyStakedTokens: Dict[str, Tuple[str, str, datetime.datetime]] = {}
-            for tokenId, ownerAddress, transactionHash, blockDate in stakedTokens:
-                currentlyStakedTokens[tokenId] = (ownerAddress, transactionHash, blockDate)
-            for tokenId, _, _, blockDate in unStakedTokens:
-                if currentlyStakedTokens[tokenId][2] < blockDate:
-                    del currentlyStakedTokens[tokenId]
-            retrievedStakedTokens += [RetrievedTokenStaking(registryAddress=registryAddress, tokenId=tokenId, stakingAddress=stakingAddress, ownerAddress=ownerAddress, stakedDate=stakedDate, transactionHash=transactionHash) for tokenId, (ownerAddress, transactionHash, stakedDate) in currentlyStakedTokens.items()]
-        return retrievedStakedTokens
+    async def retrieve_owner_token_stakings(self, ownerAddress: str, stakingAddress: str) -> List[RetrievedTokenStaking]:
+        retrievedTokenStakings: List[RetrievedTokenStaking] = []
+        if stakingAddress == CREEPZ_STAKING_ADDRESS:
+            stakedTokensResponse = (await self.ethClient.call_function(toAddress=stakingAddress, contractAbi=self.creepzStakingContractAbi, functionAbi=self.creepzStakingGetStakerTokenFunctionAbi, arguments={'staker': ownerAddress}))
```

Comment:
this (doing it by owner) is definitely not a good approach - its super duper specific to creepz. why was the flow you had before not appropriate? i.e. when something is transferred to the staking address then **just for that token** we run this function? and when we do a collection we just run the same function for all the tokens where the owner is the staking address?